subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/JUnitFilteringIntegrationTest.groovy

```diff
@@ -182,4 +184,60 @@ class JUnitFilteringIntegrationTest extends AbstractTestFilteringIntegrationTest
         result.testClass("FooServerTest").assertTestCount(1, 0, 0);
         result.testClass("FooServerTest").assertTestsExecuted("testFooServer")
     }
+
+    @Unroll
+    def 'filter as many classes as possible before sending to worker process'() {
+        given:
+        Assume.assumeFalse(framework == "JUnitPlatform") // JUnitPlatformTestClassProcessor won't emit test suite events if they're excluded
```

Comment:
❌ This comment doesn't really help me understand the problem. Why do I need suite events for this test?